code/modules/medical/surgery_tools.dm

```diff
@@ -691,29 +691,17 @@ TYPEINFO(/obj/machinery/defib_mount)
 
 		UpdateIcon()
 
-		//set move callback (when user moves, defib go back)
-		if (islist(user.move_laying))
-			user.move_laying += src
-		else
-			if (user.move_laying)
-				user.move_laying = list(user.move_laying, src.defib)
-			else
-				user.move_laying = list(src.defib)
-
```

Comment:
Why remove this? It's archaic but it works.

Reply:
I tested it with and without this piece of code. With it, the defib would sometimes teleport back to the mount even if you don't move from your tile at all, just turn or walk into a wall. Granted, it makes it usually snap back almost instantly if you walk away, but I felt it'd probably cause more confusion than anything.

I also looked a lot at the phone handset code for how that was dealt with, and it didn't have this kind of code either, only code under process(), so I felt it was fine to do the same here.

Reply:
Process tends to be a lot less reliable on live and can lead to some weird inconsistencies, would be good to register to `user`'s `COMSIG_MOVABLE_MOVED` and then call back to a range check. (though if you don't know how to do that, this works well enough)

Reply:
That was originally what I tried, but I couldn't quite get it working nicely and got lazy. I can look into it again though.

Reply:
hmm yeah I'm not sure how to get it working nicely if people drop and pull it or throw it to someone else

Reply:
okay, converted to signals, they are lovely, but idk if im doing stuff right at all, very anxiously pushed commit
works perfectly in game though, very snappy, even works with throws

Reply:
this isn't ideal, since if person A grabs it from the mount and drops it, person B can pick it up from the floor and just walk away with it, but I don't have brainpower currently to figure out how to fix

Reply:
I'd probably just keep the process part as well

Reply:
you know what, that lets me be lazy, very good idea

Reply:
okay everything works now! ideally we'd probably have a component that handles snapping back or something and use it for everything, but that's not my problem 😎